auditlog/models/rule.py

```diff
@@ -473,44 +473,69 @@ def create_logs(
         if new_values is None:
             new_values = EMPTY_DICT
         log_model = self.env["auditlog.log"]
-        http_request_model = self.env["auditlog.http.request"]
-        http_session_model = self.env["auditlog.http.session"]
-        for res_id in res_ids:
-            model_model = self.env[res_model]
-            name = model_model.browse(res_id).name_get()
-            model_id = self.pool._auditlog_model_cache[res_model]
-            auditlog_rule = self.env["auditlog.rule"].search(
-                [("model_id", "=", model_id)]
-            )
-            res_name = name and name[0] and name[0][1]
-            vals = {
-                "name": res_name,
-                "model_id": self.pool._auditlog_model_cache[res_model],
-                "res_id": res_id,
-                "method": method,
-                "user_id": uid,
-                "http_request_id": http_request_model.current_http_request(),
-                "http_session_id": http_session_model.current_http_session(),
-            }
-            vals.update(additional_log_values or {})
-            log = log_model.create(vals)
+        current_http_request = self.env["auditlog.http.request"].current_http_request()
+        current_http_session = self.env["auditlog.http.session"].current_http_session()
+
+        model_id = self.pool._auditlog_model_cache[res_model]
+        auditlog_rule = self.env["auditlog.rule"].search(
+            [("model_id", "=", model_id)],
+            limit=1,
+        )
+        log_rules_values_to_create = []
+        for batched_res_ids in self._cr.split_for_in_conditions(res_ids):
+            names = self.env[res_model].browse(batched_res_ids).name_get()
+            for name, res_id in zip(names, batched_res_ids):
+                res_name = name and name[1]
```

Comment:
For this to work with the `zip` and all, it would have to be the case that name_get returns an empty tuple or something instead of the regular (id, name) format. Does that really happen?